deployments/dex.go

```diff
@@ -0,0 +1,289 @@
+		return nil
+	}
+
+	message := "Removing helm release " + DexDeploymentID
+	out, err := helpers.WaitForCommandCompletion(ui, message,
+		func() (string, error) {
+			return helpers.RunProc(currentdir, k.Debug,
+				"helm", "uninstall", DexDeploymentID, "--namespace", DexDeploymentID)
+		},
+	)
+	if err != nil {
+		if strings.Contains(out, "release: not found") {
+			ui.Exclamation().Msgf("%s helm release not found, skipping.\n", DexDeploymentID)
+		} else {
+			return errors.Wrapf(err, "Failed uninstalling helm release %s: %s", DexDeploymentID, out)
+		}
+	}
+
+	message = "Deleting Dex namespace " + DexDeploymentID
+	_, err = helpers.WaitForCommandCompletion(ui, message,
+		func() (string, error) {
+			return "", c.DeleteNamespace(ctx, DexDeploymentID)
+		},
+	)
+	if err != nil {
+		return errors.Wrapf(err, "Failed deleting namespace %s", DexDeploymentID)
+	}
+
+	err = c.WaitForNamespaceMissing(ctx, ui, DexDeploymentID, k.Timeout)
+	if err != nil {
+		return errors.Wrap(err, "failed to delete namespace")
+	}
+
+	ui.Success().Msg("Dex removed")
+
+	return nil
+}
+
+func (k Dex) apply(ctx context.Context, c *kubernetes.Cluster, ui *termui.UI, options kubernetes.InstallationOptions, upgrade bool) error {
+	action := "install"
+	if upgrade {
+		action = "upgrade"
+	}
+
+	if err := c.CreateNamespace(ctx, DexDeploymentID, map[string]string{
+		kubernetes.EpinioDeploymentLabelKey: kubernetes.EpinioDeploymentLabelValue,
+	}, map[string]string{"linkerd.io/inject": "enabled"}); err != nil {
+		return err
+	}
+
+	domain, err := options.GetString("system_domain", TektonDeploymentID)
+	if err != nil {
+		return errors.Wrap(err, "Couldn't get system_domain option")
+	}
+
+	issuer := options.GetStringNG("tls-issuer")
+
+	// Wait for the cert manager to be present and active. It is required
+	if err := waitForCertManagerReady(ctx, ui, c, issuer); err != nil {
+		return errors.Wrap(err, "waiting for cert-manager failed")
+	}
+
+	currentdir, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+
+	tarPath, err := helpers.ExtractFile(dexChartFile)
+	if err != nil {
+		return errors.New("Failed to extract embedded file: " + tarPath + " - " + err.Error())
+	}
+	defer os.Remove(tarPath)
+
+	// We wait for Traefik Forward Auth client secret
+	// to be created so that we can use the values of the
+	// secret in the dex config.
+	if err := c.WaitForNamespace(ctx, ui, TraefikForwardAuthDeploymentID, duration.ToDeploymentNamespaceCreated()); err != nil {
+		return errors.Wrapf(err, "waiting for %s deployment", TraefikForwardAuthDeploymentID)
+	}
+	traefikAuthClientSecretName := fmt.Sprintf("%s-client", TraefikForwardAuthDeploymentID)
+	traefikAuthClientSecret, err := c.WaitForSecret(ctx, TraefikForwardAuthDeploymentID, traefikAuthClientSecretName, duration.ToAuthClientSecretCreated())
+	if err != nil {
+		return errors.Wrapf(err, "waiting for %s secret", traefikAuthClientSecretName)
+	}
+
+	apiUserOpt, err := options.GetOpt("user", "")
```

Comment:
You can use `GetString(string, string) (string, error)` to avoid the cast or `GetStringNG(string) string` to just get the value.

Reply:
Oh, right! I was copying from [the epinio deployment](https://github.com/epinio/epinio/blob/main/deployments/epinio.go#L112-L125). I will fix it there too.